### PR TITLE
Add default factory and improve initialState

### PR DESCRIPTION
**Pull request by thoov**
@stefanpenner

### CODE REVIEW

File: addon/state-for.js [hunk 1]
```diff
@@ -1,16 +1,16 @@
 import Ember from 'ember';
 import WeakMap from 'ember-weakmap/weak-map';
 
-var {
+let {
   computed,
-  assert
+  assert,
```
Comment (thoov) [inline addon/state-for.js]: I need to remove this

File: addon/state-for.js [hunk 2]
```diff
@@ -61,16 +61,16 @@ export default function stateFor(stateName, propertyName) {
...
-  let defaultState  = {};
-  let containerName = `state:${stateName}`;
-  let StateFactory  = container.lookupFactory(containerName);
+  let defaultState    = {};
+  const containerName = `state:${stateName}`;
+  let StateFactory    = container.lookupFactory(containerName);
 
   if (!StateFactory) {
-    throw new TypeError(`Unknown StateFactory: ${containerName}`);
+    StateFactory = {};
```
Comment (thoov) [inline addon/state-for.js]: What should the default state be? a blank POJO or an ember object? @stefanpenner

Comment (stefanpenner) [inline addon/state-for.js]: lets start with and Ember.Object, that way get/set do what one expects.

File: tests/unit/state-service-test.js [hunk 3]
```diff
@@ -100,3 +100,40 @@ test('that state can be a simple POJO', function(assert) {
...
+  }).create();
+
+  assert.expect(2);
+  mockObject.get('data');
+});
+
+test('that if the state name does not exist one will be proved for you', function(assert) {
+  let mockObject = Ember.Object.extend({
+    container,
+    model: mockModelA,
```
Comment (stefanpenner) [inline tests/unit/state-service-test.js]: should likely be passed to `create`, doesn't really make sense on the prototype

File: tests/unit/state-service-test.js [hunk 4]
```diff
@@ -100,3 +100,40 @@ test('that state can be a simple POJO', function(assert) {
...
+      done();
+      return {};
+    }
+  });
+
+  registry.register('state:foo-test-state', mockStateFoo, registryOpts);
+
+  mockObject = Ember.Object.extend({
+    container,
+    model: mockModelA,
```
Comment (stefanpenner) [inline tests/unit/state-service-test.js]: ^

File: README.md [hunk 5]
```diff
@@ -82,7 +82,7 @@ import Ember from 'ember';
 var MyStateObject = Ember.Object.extend();
 
 MyStateObject.reopenClass({
-  initialState() {
+  initialState(consumerContext) {
```
Comment (stefanpenner) [inline README.md]: `consumerContext` kinda sucks as a name, not really sure what to call it though...

Comment (thoov) [inline README.md]: Ya I feel the same way. I was hoping you hand some ideas :-)

File: addon/state-for.js (hunk 2 above)
Comment (thoov) [inline addon/state-for.js]: Ok I have it right now as an Ember.Object

**Comment by stefanpenner** [comment]
So the argument is likely a stateConsumer, consumer, instance.

I think it's hard to name. Instance is likely good enough. Thoughts?

**Comment by thoov** [comment]
@stefanpenner I switched to instance as that so far is the winner but Im still not completely sold on it.

**Comment by stefanpenner** [comment]
Ya, unsure what else to call it. I suspect a brief snippet in the docs describing what the instance it...

**Pull request merged**